### PR TITLE
User settings layout 

### LINKED ISSUE #231: [FEAT] - User Dashboard Settings Side nav
### Description:

Implement the User Dashboard Settings Side nav. The Nav Bar should adhere strictly to the design specifications provided. Below are the detailed instructions and requirements.

**Requirements:**

1. **UI Components:**
   - Use ShadCN UI components for consistency.
   - Ensure components match the design specifications provided.

2. **Design Specifications:**
   - Refer to the provided design for dimensions, colors, and layout.
   - Include all navigation tabs and drop down as designed.

3. **Functionality:**
   - Make sure all navigation tabs route to their specified pages 

4. **Accessibility:**
   - Ensure components is accessible and follow best practices 

5. **Testing:**
   - Ensure the component renders correctly with all required elements.

**Deadline: Friday 4pm** 
    
**Acceptance Criteria:**

- [ ] The `UserDashboardSettingsSideNavBar` component is created and located in the `components` directory.
- [ ] The Side Nav Bar matches the design specifications.
- [ ] The Side Nav Bar functions correctly and includes all items as per the design.
- [ ] Components are accessible and follow best practices.

**Attachments:**
Screenshot 2024-07-18 at 17 55 46

[Link to Design File](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2179-37386&m=dev)

- [ShadCN UI Documentation](https://ui.shadcn.com/)

### PR DESCRIPTION
**Pull request by phoenixdahdev**

### CODE REVIEW

File: tailwind.config.ts [hunk 1]
```diff
@@ -20,12 +20,14 @@ const config = {
...
         border: "hsl(var(--border))",
+        "input-border": "#CBD5E1",
+        "sidebar-background": "#FAFAFA",
         input: "hsl(var(--input))",
         ring: "hsl(var(--ring))",
         background: "hsl(var(--background))",
         foreground: "hsl(var(--foreground))",
         primary: {
-          DEFAULT: "hsl(var(--primary))",
+          DEFAULT: "#F97316",
```
Comment (ickynavigator) [inline tailwind.config.ts:30]: don't change the default color (at least not from here)

File: tailwind.config.ts [hunk 2]
```diff
@@ -20,12 +20,14 @@ const config = {
     extend: {
       colors: {
         border: "hsl(var(--border))",
+        "input-border": "#CBD5E1",
+        "sidebar-background": "#FAFAFA",
```
Comment (ickynavigator) [inline tailwind.config.ts:24]: remove these colors from here (you can add them ass variables in the app if there's actually a need)

File: src/types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { type Icon } from "lucide-react";
+
+export type SettingsSidebarProps = {
+  id?: number;
+  label: string;
+  icon: Icon;
+  link: string;
+};
```
Comment (ickynavigator) [inline src/types/index.d.ts:8]: no need to decouple the types. place them beside the constant that they are defining.

also you don't need a type for this (you can just leave the constant be)

File: src/lib/constant.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import { Banknote, Bell, Database, Globe, User, UserPlus } from "lucide-react";
+
+import { SettingsSidebarProps } from "~/types";
+
+export const Settings_Sidebar_Links: SettingsSidebarProps[] = [
```
Comment (ickynavigator) [inline src/lib/constant.ts:5]: move this into the sidebar file (we don't have any component using this)

File: src/components/common/settings/sidebar.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+const SettingsSidebar = () => {
+  const [activeLink, setActiveLink] = useState("");
+  const router = useRouter();
+  const pathname = usePathname();
+  // remove the / from the pathname
+  const currentPath = pathname?.replace("/settings/", "");
+
+  useEffect(() => {
+    setActiveLink(currentPath);
+  }, [currentPath]);
```
Comment (ickynavigator) [inline src/components/common/settings/sidebar.tsx:20]: looks like active link is just the currentPath so we can actually use that directly and avoid having to sync it

File: src/components/common/settings/sidebar.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            href={`/settings/${link.link}`}
+            key={link.id}
+            aria-current={activeLink === link.link ? "page" : undefined}
+            onKeyUp={(t) => {
+              if (t.key === "Enter" || t.key === " ") {
+                setActiveLink(link.link);
+                return;
+              }
+            }}
+            tabIndex={0}
```
Comment (ickynavigator) [inline src/components/common/settings/sidebar.tsx:46]: maybe you actually want a button 👀

File: src/components/common/settings/sidebar.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+            aria-label={link.label}
```
Comment (ickynavigator) [inline src/components/common/settings/sidebar.tsx:47]: this is not a valid label for the aria

**Review by ickynavigator — CHANGES_REQUESTED** [review]

File: src/components/common/settings/sidebar.tsx (hunk 5 above)
Comment (phoenixdahdev) [reply src/components/common/settings/sidebar.tsx:20]: reason for this

```
`  // remove the /settings/ from the pathname
  const currentPath = pathname?.replace("/settings/", "");

  useEffect(() => {
    setActiveLink(currentPath);
  }, [currentPath]);`
```

**Pull request closed without merging**